File: src/main/java/com/deal/bookapi/entity/Book.java
package com.deal.bookapi.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "livro")
@Getter
@Setter
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int Id;
    @Column(name = "nome")
    private String name;
    @Column(name = "preco")
    private float price;
    @Column(name = "paginas")
    private int pages;
    @Column(name = "codigo")
    private int code;
    @ManyToOne
    @JoinColumn(name = "id_categoria")
    private Category category;

}

File: Deal/Livraria AspNET/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LivrariaDotnet.Models
{
    [Table("livro")]
    public class Book
    {
        [Column("id")]
        [Key]
        public int Id { get; set; }

        [Column("codigo")]
        [Required(ErrorMessage="Este campo é obrigatório")]
        public int Codigo { get; set; }

        [Column("paginas")]
        [Required(ErrorMessage="Este campo é obrigatório")]
        public int Paginas { get; set; }
        
        [Column("nome")]
        [MaxLength(50)]
        [Required(ErrorMessage="Este campo é obrigatório")]
        public string Nome { get; set; }
        
        [Column("preco")]
        [Required(ErrorMessage="Este campo é obrigatório")]
        public decimal Preco { get; set; }

        [Column("id_categoria")]
        public int CategoriaId { get; set; }
        public Category Categoria { get; set; }
    }
}